common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -524,36 +538,69 @@ public boolean removeAccount(final String environment,
                         homeAccountId,
                         realm
                 ))) {
-            return false;
+            return new AccountRecordDeletionResult(null);
         }
 
+        // If no realm is provided, remove the Account/Credetials from all realms.
+        final boolean isRealmAgnostic = null == realm;
```

Comment:
this may be a little too cute.  I'd consider using parentheses around (null==realm) to improve readability.

Reply:
Sounds good